test/unit/forms/declaration/DepartureTransportSpec.scala

```diff
@@ -50,94 +51,233 @@ class DepartureTransportSpec extends FormSpec {
 
     "has errors" when {
 
-      "mandatory field are empty" in {
+      val transportType = IMOShipIDNumber
+      val transportTypeField = DepartureTransport.meansOfTransportOnDepartureTypeKey
+      val idNumberField = s"meansOfTransportOnDepartureIDNumber_$IMOShipIDNumber"
+
+      "transport type not selected" in {
+
+        val incorrectForm =
+          Map(transportTypeField -> "", idNumberField -> "")
+
+        val result = form.bind(incorrectForm)
+        val errorKeys = result.errors.map(_.key)
+        val errorMessages = result.errors.map(_.message)
+
+        errorKeys must be(List(transportTypeField))
+        errorMessages must be(List("declaration.transportInformation.meansOfTransport.departure.error.empty"))
+      }
+
+      "transport id not provided" in {
+
+        val incorrectForm =
+          Map(transportTypeField -> IMOShipIDNumber, idNumberField -> "")
+
+        val result = form.bind(incorrectForm)
+        val errorKeys = result.errors.map(_.key)
+        val errorMessages = result.errors.map(_.message)
+
+        errorKeys must be(List(idNumberField))
+        errorMessages must be(List("declaration.transportInformation.meansOfTransport.reference.error.empty"))
+      }
+
+      "fields are incorrect" in {
+
+        val incorrectForm = Map(transportTypeField -> "incorrect", idNumberField -> "correct")
+
+        val result = form.bind(incorrectForm)
+        val errorKeys = result.errors.map(_.key)
+        val errorMessages = result.errors.map(_.message)
+
+        errorKeys must be(List(transportTypeField))
+        errorMessages must be(List("declaration.transportInformation.meansOfTransport.departure.error.incorrect"))
+      }
+
+      "means of transport on departure id number is empty" in {
```

Comment:
Isn't this test the same as "transport id not provided" ?
It looks like they're testing the same scenario but in a slightly different way.

Reply:
removed duplicate